app/controllers/orders_controller.rb

```diff
@@ -1,6 +1,10 @@
 class OrdersController < ApplicationController
   def index
-    @orders = current_user.orders
+    if current_user
+      @orders = current_user.orders
+    else
+      render file: 'public/404' unless current_admin?
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.